src/openrct2/world/ConstructionClearance.cpp

```diff
@@ -108,7 +108,9 @@ static bool MapLoc68BABCShouldContinue(
 
         const TrackElement* const trackElement = tileElement->AsTrack();
         const auto& ted = OpenRCT2::TrackMetaData::GetTrackElementDescriptor(trackElement->GetTrackType());
-        const auto& trackClearances = ted.sequences[trackElement->GetSequenceIndex()].clearance;
+        // mazes reuse the sequence index, take min of sequence index and track element sequence count
+        const uint8_t sequenceIndex = std::min<uint8_t>(trackElement->GetSequenceIndex(), ted.numSequences - 1);
+        const auto& trackClearances = ted.sequences[sequenceIndex].clearance;
```

Comment:
This still feels hacky to me. We know the ID of the maze piece, so we can just make sure it’s set to `0` in that case.

```suggestion
        const uint8_t sequenceIndex = (trackElement->GetTrackType() == TrackElemType::maze) ? 0 : trackElement->GetSequenceIndex());
        const auto& trackClearances = ted.sequences[sequenceIndex].clearance;
```